app/lib/features/bluetooth/backend/bluetooth_low_energy/ble_device.dart

```diff
@@ -0,0 +1,138 @@
+part of 'ble_manager.dart';
+
+/// BluetoothDevice implementation for the 'bluetooth_low_energy' package
+final class BluetoothLowEnergyDevice
+  extends BluetoothDevice<BluetoothLowEnergyManager, BluetoothLowEnergyService, BluetoothLowEnergyCharacteristic, DiscoveredEventArgs> {
+  /// constructor
+  BluetoothLowEnergyDevice(super._manager, super._source);
+
+  @override
+  String get deviceId => source.peripheral.uuid.toString();
+
+  @override
+  String get name => source.advertisement.name ?? deviceId;
+
+  CentralManager get _cm => manager.backend;
+
+  @override
+  Stream<BluetoothConnectionState> get connectionStream => _cm.connectionStateChanged.transform(
+    BluetoothConnectionStateStreamTransformer(stateParser: BluetoothLowEnergyConnectionStateParser())
+  );
+
+  @override
+  Future<void> backendConnect() => _cm.connect(source.peripheral);
+
+  @override
+  Future<void> backendDisconnect() => _cm.disconnect(source.peripheral);
+
+  @override
+  Future<List<BluetoothLowEnergyService>?> discoverServices() async {
+    if (!isConnected) {
+      logger.finer('discoverServices: device not connect. Call device.connect() first');
+      return null;
+    }
+
+    // Query actual services supported by the device. While they must be
+    // rediscovered when a disconnect happens, this object is also recreated.
+    try {
+      final allServices = await _cm.discoverGATT(source.peripheral);
+      logger.finer('discoverServices: $allServices');
+
+      return allServices.map(BluetoothLowEnergyService.fromSource).toList();
+    } catch (e) {
+      logger.shout('discoverServices: error:', [source.peripheral, e]);
+      return null;
+    }
+  }
+  
+  @override
+  Future<bool> getCharacteristicValueByUuid(BluetoothLowEnergyCharacteristic characteristic, List<Uint8List> value) async {
+    if (!isConnected) {
+      logger.finer('getCharacteristicValueByUuid: device not connect. Call device.connect() first');
```

Comment:
When this hints at an developer error asserts make sure it is caught.
```suggestion
      assert(false, 'Device not connect.Call device.connect() first.');
```